src/csharp/Microsoft.Spark/ML/Feature/HashingTF.cs

```diff
@@ -0,0 +1,183 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Interop.Ipc;
+using Microsoft.Spark.Sql;
+using Microsoft.Spark.Sql.Types;
+
+namespace Microsoft.Spark.ML.Feature
+{
+    /// <summary>
+    /// A <see cref="HashingTF"/> Maps a sequence of terms to their term frequencies using the
+    /// hashing trick. Currently we use Austin Appleby's MurmurHash 3 algorithm
+    /// (MurmurHash3_x86_32) to calculate the hash code value for the term object. Since a simple
+    /// modulo is used to transform the hash function to a column index, it is advisable to use a
+    /// power of two as the numFeatures parameter; otherwise the features will not be mapped evenly
+    /// to the columns.
+    /// </summary>
+    public class HashingTF : IJvmObjectReferenceProvider
+    {
+        /// <summary>
+        /// Create a <see cref="HashingTF"/> without any parameters
+        /// </summary>
+        public HashingTF()
+        {
+            _jvmObject = SparkEnvironment.JvmBridge.CallConstructor(_javaClassName);
+        }
+
+        /// <summary>
+        /// Create a <see cref="HashingTF"/> with a UID that is used to give the
+        /// <see cref="HashingTF"/> a unique ID
+        /// <param name="uid">unique identifier</param>
+        /// </summary>
+        public HashingTF(string uid)
+        {
+            _jvmObject = SparkEnvironment.JvmBridge.CallConstructor(_javaClassName, uid);
+        }
+        
+        internal HashingTF(JvmObjectReference jvmObject)
+        {
+            _jvmObject = jvmObject;
+        }
+
+        private readonly JvmObjectReference _jvmObject;
+        private const string _javaClassName = "org.apache.spark.ml.feature.HashingTF";
```

Comment:
`JavaClassName`

Reply:
Actually, to be consistnet, can this be `private static readonly string s_hashingTFClassName = "blah"`
You can search the codebase with `ClassName =` to see examples.